File: app/src/main/java/me/echeung/moemoekyun/api/endpoints/Library.java
package me.echeung.moemoekyun.api.endpoints;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public abstract class Library {

    public static final String API_BASE = "https://listen.moe/api/";

    public static final String CDN_ALBUM_ART_URL = "https://cdn.listen.moe/covers/";
    public static final String CDN_AVATAR_URL = "https://cdn.listen.moe/avatars/";
    public static final String CDN_BANNER_URL = "https://cdn.listen.moe/banners/";

    @Getter
    private final String name;
    @Getter
    private final String socketUrl;
    @Getter
    private final String streamUrl;

}

File: app/src/main/kotlin/me/echeung/moemoekyun/client/api/Library.kt
package me.echeung.moemoekyun.client.api

enum class Library(val socketUrl: String, val streamUrl: String) {
    jpop(
        "wss://listen.moe/gateway_v2",
        "https://listen.moe/fallback"
    ),
    kpop(
        "wss://listen.moe/kpop/gateway_v2",
        "https://listen.moe/kpop/fallback"
    ),
    ;

    companion object {
        const val API_BASE = "https://listen.moe/graphql"

        const val CDN_ALBUM_ART_URL = "https://cdn.listen.moe/covers/"
        const val CDN_AVATAR_URL = "https://cdn.listen.moe/avatars/"
        const val CDN_BANNER_URL = "https://cdn.listen.moe/banners/"
    }
}
